### ISSUE TITLE
[BUG] Search does not work on Elegant 5.4.0 and Pelican 4.7.1

### ISSUE THREAD

**Issue by silverhook** [Pelican-Elegant/elegant#789]
## Mandatory Step

- [x] I am using latest production release of Elegant

## Recommended Step

- [ ] I have attached link to my blog source code repository, so that developers can try to reproduce the issue at the end
- [ ] I can't my source code but I have linked my pelican configuration files to help developers reproduce the issue

## Description

**Describe the bug**

Search does not work any more. For an example with screenshots see #788.

Pages I already identified do not work right now:
- https://elegant.oncrashreboot.com – the project’s own homepage and documentation
- https://matija.suklje.name – my homepage
-  https://iranzo.io – @iranzo ’s hompeage

Interestingly enough, @talha131 ’s homepage works:
https://www.oncrashreboot.com/

**To Reproduce**

1. Enter text into the search bar
2. Press `Enter` or click the search button
3. See empty page, not even an error notice

**Expected behavior**
List of search results, as before

**Screenshots**

see #788 

**Desktop (please complete the following information):**

- OS: Linux
- Browser: Firefox
- Version: 95.0.2

**Additional context**

Relevant issue/comment: https://github.com/Pelican-Elegant/elegant/issues/628#issuecomment-678680230

https://blog.getpelican.com/pelican-4.5-released.html

[`tipue_search`](https://github.com/pelican-plugins/tipue-search) is now deprecated in Pelican in favour of [`search`](https://github.com/pelican-plugins/search) (which uses [Stork](https://stork-search.net/)).

**Comment by iranzo** [comment]
I don't like the 'download stork' approach for this tbh :/

**Comment by silverhook** [comment]
We could still continue to roll our own, but then it needs to work. It currently doesn’t.

I don’t have strong feelings one way or another which solution is better suited, as long as it’s lean enough to run on my ARM server and works :smile_cat:

**Comment by justinmayer** [comment]
@iranzo said:

> I don't like the 'download stork' approach for this tbh :/

I agree that extra step is cumbersome, but in my estimation the trade-off is well worth it. Stork is written in Rust and is therefore very fast, and above all, it is well-supported by the author.

To me, the more salient question is, "What are the alternatives?" I don't think a long-abandoned and unmaintained project is a sustainable solution, and I don't see a queue of folks lining up to identify, evaluate, select, and implement a viable alternative that provides search functionality.

If you want to see the user experience for yourself, you can see the new Stork-powered Search plugin in action at:

* [Justin Mayer](https://justinmayer.com)
* [Open Source Alternatives](https://opensourcealternatives.org)

**Comment by FrederikLauber** [comment]
Any news on this?
Trying to port an old jekyll site to pelican with elegant and the stork based search does not yet seem to work nor can I find any documentation to fix this.

**Comment by justinmayer** [comment]
Hi @FrederikLauber. I can empathize with your desire to achieve your desired outcome. That said, please refer to: [“Any Updates?”](https://justinmayer.com/posts/any-updates/)

**Comment by FrederikLauber** [comment]
@justinmayer thanks for your reply. Your link was very enlightening. I figured it out myself besides some graphic issues.

**Comment by silverhook** [comment]
@FrederikLauber, care to share more?

Search would be nice to have.

**Comment by FrederikLauber** [comment]
@silverhook 

I pushed my changes here: https://github.com/FrederikLauber/elegant

Probably not good enough for a merge request as I have no clue what I may have broken in the process but it produces a workign search.

**Comment by silverhook** [comment]
@FrederikLauber  is this your website, so I can see how the search works there?

https://flambda.de/

**Comment by FrederikLauber** [comment]
Yes it is. The design is the one from the git repo and then I just use some plugins for pelican.

**Comment by silverhook** [comment]
Sorry for the late reply, this somehow flew under my radar.

At the lack of a better option, I think that’s an OK solution. I don’t exactly like the style, but IMHO it’s better to have a working maintained search than a prettier non-working one.

Could you provide a PR, @FrederikLauber and we can perhaps polish it up together?

**Comment by FrederikLauber** [comment]
@silverhook I extracted the changes out of my fork and created a pull request here: https://github.com/Pelican-Elegant/elegant/pull/910

It probably needs a bit mire refining though from a person that knows more about elegant.

**Comment by silverhook** [comment]
Awesome, thanks, @FrederikLauber .

Unfortunately from all the Elegant “devs”, I’m the one with by far the least development skills.

Can someone else take a look, please?

**Comment by silverhook** [comment]
There is also a bug in Pelican open to replace the – since 2023 abandoned – Stork with something else:
https://github.com/pelican-plugins/search/issues/35

It seems https://pagefind.app/ may be a good contender.
